tests/e2e/test_testing.py

```diff
@@ -11,7 +11,7 @@
 def test_basic_contract(protostar):
     result = protostar(["test", "tests"])
     assert "1 passed" in result
-    assert "Seed:" not in result
+    assert "Seed:" in result
```

Comment:
Should we display seed if there are no fuzz tests? IMO this assertion should be in a test case that checks fuzzing.

Reply:
The whole point of this PR is to display the seed if there are no tests 😅 

Previously we did so, but such flag required use of shared memory objects which broke when we started passing these in a bazilion of dataclasses that we have (state, config, worker args etc.). The change in this test reflects the change in behaviour -- the seed will always be displayed now, because I have simply removed that shared memory usage entirely.

There is also an assertion for this in fuzz related e2e tests. It's not changed so it's not visible in PR diff.

Reply:
The point of this PR is fixing deadlock. Displaying seed is a side-effect, that's why I don't understand the purpose of this assertion.

Reply:
This assertion test the behaviour that has changed "in the past" and it guards against reverting this behaviour change accidentally.